controllers/machinehealthcheck_controller.go

```diff
@@ -306,9 +306,9 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 	return ctrl.Result{}, nil
 }
 
-// PatchHealthyTargets patches healthy machines with MachineHealthCheckSuccededCondition.
-func (r *MachineHealthCheckReconciler) PatchHealthyTargets(ctx context.Context, logger logr.Logger, healthy []healthCheckTarget, cluster *clusterv1.Cluster, m *clusterv1.MachineHealthCheck) []error {
-	errList := []error{}
+// PatchHealthyTargets patches healthy machines with MachineHealthCheckSucceededCondition.
+func (r *MachineHealthCheckReconciler) PatchHealthyTargets(ctx context.Context, logger logr.Logger, healthy []healthCheckTarget, m *clusterv1.MachineHealthCheck) []error {
```

Comment:
This is technically a breaking change because this method is exported (which probably shouldn't be)

Reply:
@JoelSpeed Should we unexport this method and remove the Cluster parameter?

Reply:
Seems this method was added as part of #3901, which also added `PatchUnhealthyTargets`. I think it's reasonable to unexport this, but in that case, I would suggest we do the same for `PatchUnhealthyTargets` too for consistency.

Seems the method has never used the cluster parameter so +1 to dropping that.